### ISSUE TITLE
`retain` does not work,  if you only want to keep certain subfields

### ISSUE THREAD

**Issue by TobiasNx** [metafacture/metafacture-fix#145]
Retain works with root level parent fields but not with an subfield:
https://github.com/TobiasNx/fix-FunctionalReview-Testing/blob/c9d9447983fc435d01b35ac57046dc1dd3ddbb11/data/testing/retainOnlySubfield

```
copy_field("node.properties.ccm:educationallearningresourcetype[]", "lrt[]")
retain("node.properties.ccm:educationallearningresourcetype[]", "lrt[]")
```
is:

```
{
  "lrt" : [ "https://w3id.org/kim/hcrt/image" ]
}
```

expected:
```

{
  "lrt" : [ "https://w3id.org/kim/hcrt/image" ],
  "node": {
    "properties": {
        "ccm:educationallearningresourcetype": [
            "https://w3id.org/kim/hcrt/image"
        ]
    }
}
}
```

**Comment by TobiasNx** [comment]
retain would be a clever element to learn fix-path since you would only keep certain elements.

would be a nice feature though.

**Comment by blackwinter** [comment]
Integration test:

- [retainCertainSubfields](https://github.com/metafacture/metafacture-fix/tree/master/metafix/src/test/resources/org/metafacture/metafix/integration/record/fromJson/toJson/retainCertainSubfields)

**Comment by blackwinter** [comment]
@TobiasNx: Implemented in #329. Please test.

**Comment by TobiasNx** [comment]
https://github.com/metafacture/metafacture-fix/pull/329#issuecomment-1801440592
